Expensify/App
# Ensure accounts with a private domain secondary login can create a workspace

### Details
Ensure accounts with a private domain secondary login can create a workspace via the "Get Started" button in workspace configuration.

Related Web-E PR: https://github.com/Expensify/Web-Expensify/pull/31862

cc @roryabraham 

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/176225

### Tests
1. Create an account with a public domain. e.g. `@gmail.com`.
2. Login to NewDot and navigate to `+ > New Workspace` and verify that you see `Add Email` in the card page.
3. Click on `Add email`.
4. Add a private domain secondary login to your account, e.g. `@expensify.com`.
5. Validate the secondary login by following the instructions in [this SO](https://stackoverflow.com/c/expensify/questions/304/441#441) to generate a link. **Note:** Don't use the clitools script! It does not hit the API endpoint and will not trigger the pusher event to update NewDot.
5. Go back to NewDot and verify that the button now says `Get Started` and clicking it starts the VBA flow.

### QA Steps
Steps above.

### Tested On


### Screenshots


#### Web

https://user-images.githubusercontent.com/22219519/132928039-3c8d077b-28db-4241-8272-73f218cc7433.mov

#### Mobile Web


#### Desktop

https://user-images.githubusercontent.com/22219519/132928263-4b75c156-900c-4426-898e-be7d74143f4f.mov

#### iOS


#### Android
